apps/block_scout_web/assets/js/lib/smart_contract/connect.js

```diff
@@ -0,0 +1,177 @@
+import Web3 from 'web3'
+import Web3Modal from 'web3modal'
+import WalletConnectProvider from '@walletconnect/web3-provider'
+import { compareChainIDs, formatError, showConnectElements, showConnectedToElements } from './common_helpers'
+import { openWarningModal } from '../modals'
+
+const instanceChainId = process.env.CHAIN_ID ? parseInt(`${process.env.CHAIN_ID}`, 10) : 100
+const walletConnectOptions = { rpc: {}, chainId: instanceChainId }
+walletConnectOptions.rpc[instanceChainId] = process.env.JSON_RPC ? process.env.JSON_RPC : 'https://dai.poa.network'
```

Comment:
We may have to add this to blockscout environment in order to use wallet connect stuff

Reply:
Updated the default value just in case.